ppsci/arch/meteoformer.py

```diff
@@ -0,0 +1,435 @@
+        B, T, C, H, W = x.shape
+        # B TC H W
+
+        x = x.reshape([B, T * C, H, W])
+        # B HW TC
+        x = x.flatten(2).transpose(perm=[0, 2, 1])
+
+        # encoder
+        z = x
+        for i in range(self.N_T):
+            z = self.enc[i](z, H, W)
+
+        return z
+
+
+# MultiDecoder
+class Decoder(nn.Layer):
+    def __init__(self, C_hid: int, C_out: int, N_S: int):
+        super(Decoder, self).__init__()
+        strides = stride_generator(N_S, reverse=True)
+        # strides = [2, 1, 2, 1]
+        self.dec = nn.Sequential(
+            *[ConvSC(C_hid, C_hid, stride=s, transpose=True) for s in strides[:-1]],
+            ConvSC(C_hid, C_hid, stride=strides[-1], transpose=True),
+        )
+        self.readout = nn.Conv2D(C_hid, C_out, 1)
+
+    def forward(self, hid, enc1=None):
+        for i in range(0, len(self.dec)):
+            hid = self.dec[i](hid)
+        Y = self.readout(hid)
+        return Y
+
+
+class Meteoformer(base.Arch):
+    """
+    Meteoformer is a class that represents a Spatial-Temporal Transformer model designed for short-to-medium-term weather prediction with multiple meteorological variables.
+
+    Args:
+        input_keys (Tuple[str, ...]): A tuple of input keys.
+        output_keys (Tuple[str, ...]): A tuple of output keys.
+        shape_in (Tuple[int, ...]): The shape of the input data (T, C, H, W), where
+            T is the number of time steps, C is the number of channels,
+            H and W are the spatial dimensions.
+        hid_S (int): The number of hidden channels in the spatial encoder.
+        hid_T (int): The number of hidden units in the temporal encoder.
+        N_S (int): The number of spatial transformer layers.
+        N_T (int): The number of temporal transformer layers.
+        incep_ker (Tuple[int, ...]): The kernel sizes used in the inception block.
+        groups (int): The number of groups for grouped convolutions.
+        num_classes (int): The number of predicted meteorological variables.
+
+    Examples:
+        >>> import paddle
+        >>> import ppsci
+        >>> model = ppsci.arch.Meteoformer(
+        ...     input_keys=("input",),
+        ...     output_keys=("output",),
+        ...     shape_in=(12, 12, 192, 256),
+        ...     hid_S=64,
+        ...     hid_T=256,
+        ...     N_S=4,
+        ...     N_T=4,
+        ...     incep_ker=(3, 5, 7, 11),
+        ...     groups=8,
+        ...     num_classes=4,
+        ... )
+        >>> input_dict = {"input": paddle.rand([16, 12, 4, 192, 256])}
+        >>> output_dict = model(input_dict)
+        >>> print(output_dict["output"].shape)
+        [16, 12, 4, 192, 256]
+    """
+
+    def __init__(
+        self,
+        input_keys: Tuple[str, ...],
+        output_keys: Tuple[str, ...],
+        shape_in: Tuple[int, ...],
+        hid_S: int = 64,
+        hid_T: int = 256,
+        N_S: int = 4,
+        N_T: int = 4,
+        incep_ker: Tuple[int, ...] = [3, 5, 7, 11],
```

Comment:
默认参数不建议使用list这种可变类型，而是tuple
```suggestion
        incep_ker: Tuple[int, ...] = (3, 5, 7, 11),
```